File: tiffany-backend/src/main/java/com/tiffany/features/auth/dto/response/SocialAuthResponse.java
package com.tiffany.features.auth.dto.response;

import com.tiffany.enums.user.UserRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SocialAuthResponse {

    private boolean success;
    private String message;
    private String provider;

    private UUID userId;
    private String userIdentifier;
    private String userType;
    private UserRole userRole;

    private String accessToken;
    private String refreshToken;
}

File: tiffany-frontend/src/redux/features/auth/store/models/response/social-auth-response.ts
/**
 * Social Authentication Response Models
 */

export type OperationType = "LOGIN" | "REGISTRATION";

/**
 * Social authentication response
 * Returned after successful Telegram/Google login or registration
 */
export interface SocialAuthResponse {
  success: boolean;
  message: string;
  provider: string;
  userId: string;
  userIdentifier: string;
  userType: string;
  accessToken: string;
  refreshToken: string;
  socialId: string;
  socialUsername: string;
  syncedAt: string;
  operationType: OperationType;
  isNewUser: boolean;
}

/**
 * Social sync response
 * Returned after connecting/disconnecting social account
 */
export interface SocialSyncResponse {
  success: boolean;
  message: string;
  provider: string;
  syncedAt: string | null;
  telegramId: number | null;
  telegramUsername: string | null;
  telegramFirstName: string | null;
  telegramLastName: string | null;
  telegramPhotoUrl: string | null;
  googleId: string | null;
  googleEmail: string | null;
}

/**
 * Refresh token response
 */
export interface RefreshTokenResponse {
  accessToken: string;
  refreshToken: string;
  tokenType: string;
}
